### ISSUE TITLE
⚙Tech Debt: Detect existing IDs

### ISSUE THREAD

**Issue by huntabyte** [melt-ui/melt-ui#686]
In addition to #683, I've been thinking about other ways we could possibly improve the DX of some builders.

What if we made the internal `ids` a store, which would enable us to automatically detect/update if an element the action is applied to has an `id` already and use that id.

This would remove some of the conflicts that occur when you want to use multiple builders on a single element, for example, a dialog trigger that also has a tooltip.

Thoughts on this @TGlide ?

**Comment by TGlide** [comment]
> What if we made the internal ids a store, which would enable us to automatically detect/update if an element the action is applied to has an id already and use that id instead of generating our own?

How would that work?

--- 

An idea popped into my head. Drafting a POC

**Comment by TGlide** [comment]
Alright, POC didn't work out 🙃 

This is a more difficult issue than it seems. The quick and dirty solution would be to have a store for each id, similar to what you said.

e.g.
```ts
const ids = generateIds('content', 'title', 'description', 'trigger');
const idsStore = toWritableStores(ids);
```

But this wouldn't make it necessarily easy to merge builders. What we could do, is have the builder accept an optional id parameter? And then have the mergeBuilder helper work with that. I'll play around with the idea

**Comment by TGlide** [comment]
Okay, with #674 merged, we can now think of a `mergeBuilders` helper, somehow.

**Comment by huntabyte** [comment]
I don't think this is a MAJOR priority at the moment given we now have the ability to add custom ids, etc. to work around some of the issues with multiple builders. It also isn't _always_ in the best interest of a11y to allow the merging of certain things.
